### ISSUE TITLE
Add HyperX Alloy Elite RGB

### ISSUE THREAD

**Issue by xscorpionxz** [antonpup/Aurora#1962]
Hi, 
OpenRGB now support this Keyboard. 
Do you think you can work around this third party SDK ?

**Comment by GreatSkull** [comment]
Well Someone is already adding it https://www.reddit.com/r/HyperX/comments/hgd1p9/alloy_elite_rgb_integration_into_aurora_software/

And I hope it works since HyperX officially dropped support for this keyboard in order to launch the HyperX Alloy Elite 2, so it would be nice that the community supports this keyboard that was supposed to be a "flagship" for HyperX and a lot of people already have it, but since HyperX is being anti-consumer and just re-releasing the same keyboard just with pudding PBT keycaps and making the non pudding keycaps obsolete so people will buy that instead.

**Comment by GreatSkull** [comment]
@xscorpionxz @diogotr7 @simon-wh 

Test this custom build
https://ci.appveyor.com/project/diogotr7/aurora/builds/33776184/artifacts

but remember install the latest openRGB job
https://gitlab.com/CalcProgrammer1/OpenRGB/-/pipelines

but follow the turtorial first https://gitlab.com/CalcProgrammer1/OpenRGB/-/wikis/OpenRGB-Windows-Setup-and-Usage

Remember to install the winUSB driver to interface 2.

And also remember that the media keys will freeze the keyboard hwn WinUSB driver is installed, devs are currently working to fix this issue

**Comment by diogotr7** [comment]
I wrote the support for openrgb, you don't need to tag me...

**Comment by GreatSkull** [comment]
> 
> 
> I wrote the support for openrgb, you don't need to tag me...

Yeah but i'm here to thank you, thanks

**Comment by Dagda52** [comment]
Thanks, is working as intended, you don't need to install the winUSB driver anymore in the latest openRGB release so the compability is great, the keyboard is not freezing anymore.

**Comment by GreatSkull** [comment]
> 
> 
> Thanks, is working as intended, you don't need to install the winUSB driver anymore in the latest openRGB release so the compability is great, the keyboard is not freezing anymore.

Yes, but the only issue that needs to be pointed is that if you switch between on-board profiles while openRGB is on is that they will be messed up and mixed with any lighting that is currently active (static only).

Since openRGB no longer needs winUSB you can easily fix that by reopening Ngenuity and re-flash the profiles again and don't switch between profiles.

You can also use ngenuity + openRGB + Aurora if you need any custom profile just remember to open Ngenuity first before openRGB/aurora or else ngenuity will not detect the keyboard.

**Comment by DirkTeulings** [comment]
How do you get this working?
Installed this beta, and does not detect keyboard.
Going into the device manager openRGB is not initialized. 

Pressing start won't do any thing too.

Note: Using latest version of openRGB (0.3)

**Comment by CalcProgrammer1** [comment]
You have to start the SDK Server in OpenRGB.

The next step here is to standardize key names across all keyboards in OpenRGB, so that Aurora can parse the keymap automatically rather than having it hard-coded for specific devices as it is in the current OpenRGB branch.

**Comment by DirkTeulings** [comment]
Correct, that works!

**Comment by CalcProgrammer1** [comment]
@diogotr7 and I worked on OpenRGB integration some more this morning.  We reworked the Aurora support to automatically map the OpenRGB LEDs based on a standardized list of key names.  I updated the OpenRGB side so that all keyboards have matching key names.  The end result is 5 keyboards all synced to Aurora:

https://gfycat.com/importantinsecuredalmatian

Mouse support is currently not implemented.  All other devices respond to the generic peripheral logo key color.  The latest code is in @diogotr7's branch:

https://github.com/diogotr7/Aurora/tree/feature/openrgb

It's working quite nicely on my setup.

Tested Keyboards for OpenRGB Integration:

* HyperX Alloy Elite
* TTEsports Poseidon Z RGB
* Logitech G810
* Corsair K70 RGB
* Razer Huntsman Elite
* Razer BlackWidow Chroma
* Razer BlackWidow Chroma TE

Edit: Added my laptops:

* Razer Blade Stealth (2016)
* Razer Blade Pro (2017)

Also added some code to OpenRGB to set high resolution timer to fix slow update rates on certain devices.

**Comment by Dagda52** [comment]
> Tested Keyboards for OpenRGB Integration:
> 
> * HyperX Alloy Elite
> * TTEsports Poseidon Z RGB
> * Logitech G810
> * Corsair K70 RGB
> * Razer Huntsman Elite
> * Razer BlackWidow Chroma
> * Razer BlackWidow Chroma TE

Nice, so we can finally ditch synapse and ngenuity?, this is a great day, thanks for this.....

**Comment by Cnuut** [comment]
Hi sorry for leeching on this thread, but will there be any love for the 'HyperX Alloy Origins'? //Jim

**Comment by CalcProgrammer1** [comment]
OpenRGB has Alloy Origins support as of 0.3.  It would be great if you could test the latest build of OpenRGB along with the latest OpenRGB Aurora branch posted above.

**Comment by Cnuut** [comment]
I followed the above installinstructions, the keyboard works fine with ORGB, but when I open Aurora the LEDs go dark (set the keyboard as 'Alloy Elite') messed around abit in Aurora and also installed a profile, but still the darkness prevails ;)

**Comment by diogotr7** [comment]
@Cnuut make sure to use this specific build:

https://ci.appveyor.com/project/diogotr7/aurora/builds/34267837

If it asks you to update, don't. I forgot to update the version so it thinks the latest stable release is more recent...

**Comment by Cnuut** [comment]
Yes! Now it works, thank you!

**Comment by rehv** [comment]
I'm using this build (https://ci.appveyor.com/project/diogotr7/aurora/builds/34267837) w/ OpenRGB 0.3, works fine (THANKS!) but the \ | (above Enter key) doesn't light up. I've already switched the preferred keyboard layout to "United States" (the image displayed matches my layout).

Anyone know what can be done?

Thanks in advance 👍 

EDIT: My KB is the Alloy Elite RGB.

**Comment by CalcProgrammer1** [comment]
Hmm, I have the US layout Alloy Elite and it was working last time I checked.  I'll have to check it again.  The key named "Key: \ (ANSI)" is the US (and other ANSI layouts) backslash above Enter.  The key named "Key: \ (ISO)" is the UK (and other ISO layouts) backslash in between left shift and Z.  The HyperX Alloy Elite keymap has both keys and as long as Aurora is set correctly I think it should work.

**Comment by rehv** [comment]
Yeah, I don't know what I can do... I can light the key with OpenRGB, but it doesn't light up in Aurora... I've tried 4 builds already, but only the ones linked here had the HyperX Alloy Elite RGB option and those won't light that key up.

When I go to profiles and specifically assign the \ | key it doesn't light up (it does recognize it when I press it, tho).
IDK what else I can do :|

**Comment by IMNdi** [comment]
Additional observation. I have the international version (\ above enter) and the key is off.

I have an interactive layer that makes pressed key red then fades it out. 

I press "\":
On-screen key "\" goes red (even if wrongly placed)
On the physical keyboard, ENTER goes red.

I press "Enter"
On-screen "Enter" goes red
Nothing on the keyboard responds. (presumably the dark key should fade?)

Does that help?

**Comment by MarcoBrieden** [comment]
I have the Razer Ornata Chroma, Logitech G502 Hero, AMD Wraith Prism and ASUS Aura GPU, all working fine in OpenRGB 0.4.
But I can't get the keyboard working in any Aurora build. It seems Aurora is constantly sending the 'turn off' signal to all Keyboard LEDs.

I can help debug, or maybe try it on my own if someone has some pointers on how the Aurora codebase works.

**Comment by CalcProgrammer1** [comment]
I need to add a keymap for the Ornata Chroma.  Mine's been at my office at work for years, but I'm going to take it home to figure out the keymap so I can have it in for OpenRGB 0.5.  Aurora relies on a keymap to control the correct keys.

**Comment by MarcoBrieden** [comment]
Any way I can help?

**Comment by CalcProgrammer1** [comment]
It should be working in the latest build of OpenRGB now.  The Ornata Chroma keymap is very similar to the Blackwidow but has a few minor differences that I had to correct.  I've tested every labeled key and it is working for me.

**Comment by IMNdi** [comment]
I allowed Aurora to update and it appears that "OpenRGB" is not in it, nor is the Alloy listed as supported that I could see.
Meanwhile, the link above no longer contains a build (I think they expire after 6 months).

Other than me restoring from a backup (I did), is this effectively dead? I can't see any way in which a new user could make this work.

**Comment by CalcProgrammer1** [comment]
You need the dev version.  The release version is effectively dead, and dev
is mostly dead.  Most development has moved over to Artemis.
